test/integ/sharedMem/src/sharedMem.cpp

```diff
@@ -160,8 +161,11 @@ TEMPLATE_LIST_TEST_CASE("sharedMem", "[sharedMem]", TestAccs)
     auto blockRetValsAcc = alpaka::allocBuf<Val, Idx>(devAcc, resultElemCount);
     alpaka::memcpy(queue, blockRetValsAcc, blockRetVals, resultElemCount);
 
-    // Create the kernel execution task.
-    auto const taskKernel = alpaka::createTaskKernel<Acc>(workDiv, kernel, std::data(blockRetValsAcc));
+
+    // Create the kernel execution task using the real data.
+    auto const& kernelBundle2 = alpaka::KernelBundle(kernel, std::data(blockRetValsAcc));
```

Comment:
I think this could be simply
```suggestion
    auto const kernelBundle2 = alpaka::KernelBundle(kernel, std::data(blockRetValsAcc));
```
since it's a constructor call ?

Reply:
However, I don't like that the bundle is created only to execute the kernel.

This is another example where I don't like forcing the use of the bundles.

Reply:
But it is required for `getValidWorkDivKernel`. We kept therefore the old interface but if you use `getValidWorkDivKernel` you should use the bundle for the kernel call to avoid signature mismatch which can end with a kernel crash due to invalid work extents.

Reply:
I understand - but we are not using `kernelBundle2` with `getValidWorkDivKernel`.

Reply:
fair point. In prinziple the example is broken because the workdiv is assumed to work for the second kernel too. I will change this line back